pyproj/transformer.py

```diff
@@ -820,6 +821,105 @@ def transform(  # pylint: disable=invalid-name
             return_data += (_convertback(t_data_type, intime),)
         return return_data
 
+    @overload
+    def transform_point(  # pylint: disable=invalid-name
+        self,
+        xx: float,
+        yy: float,
+        radians: bool = False,
+        errcheck: bool = False,
+        direction: Union[TransformDirection, str] = TransformDirection.FORWARD,
+    ) -> Tuple[float, float]:
+        ...
+
+    @overload
+    def transform_point(  # pylint: disable=invalid-name
+        self,
+        xx: float,
+        yy: float,
+        zz: float,
+        radians: bool = False,
+        errcheck: bool = False,
+        direction: Union[TransformDirection, str] = TransformDirection.FORWARD,
+    ) -> Tuple[float, float, float]:
+        ...
+
+    @overload
+    def transform_point(  # pylint: disable=invalid-name
+        self,
+        xx: float,
+        yy: float,
+        zz: float,
+        tt: float,
+        radians: bool = False,
+        errcheck: bool = False,
+        direction: Union[TransformDirection, str] = TransformDirection.FORWARD,
+    ) -> Tuple[float, float, float, float]:
+        ...
+
+    def transform_point(  # pylint: disable=invalid-name
+        self,
+        xx,
+        yy,
+        zz=None,
+        tt=None,
+        radians=False,
+        errcheck=False,
+        direction=TransformDirection.FORWARD,
+    ):
+        """
+        Optimized to transform single point between two coordinate systems.
+
+        See: :c:func:`proj_trans`
+
+        .. versionadded:: 2.5.0
+
+        Accepted numeric scalar:
```

Comment:
Do you want to say `numbers.Real` instead of explicitly listing these out? https://docs.python.org/3/library/numbers.html#numbers.Real

Reply:
Good suggestion. I added it as the type and kept the list as examples.